lib/models/wallet_connect/request_send_qubic_event.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:form_builder_validators/form_builder_validators.dart';
+import 'package:qubic_wallet/di.dart';
+import 'package:qubic_wallet/helpers/id_validators.dart';
+import 'package:qubic_wallet/models/wallet_connect/request_event.dart';
+import 'package:qubic_wallet/stores/application_store.dart';
+// ignore: depend_on_referenced_packages
+import 'package:collection/collection.dart';
+import 'package:walletconnect_flutter_v2/walletconnect_flutter_v2.dart';
+
+class RequestSendQubicEvent extends RequestEvent {
+  final String fromID; //From which publicID should the funds flow
+  final String toID; //To which publicID should the funds flow
+  final int amount; //The amount of funds to send
+
+  late final String fromIDName; //The name of the fromID
+  late final PairingMetadata?
+      pairingMetadata; //The pairing metadata to send the request
+
+  //Validates the request to send qubic against the wallet context
+  void validateOrThrow() {
+    ApplicationStore appStore = getIt<ApplicationStore>();
+    var account =
+        appStore.currentQubicIDs.firstWhereOrNull((e) => e.publicId == fromID);
+    if (account == null) {
+      throw ArgumentError("fromID is unknown");
+    }
+    if ((account.amount == null) || (account.amount! < amount)) {
+      throw ArgumentError("insufficient funds in fromID");
+    }
+    if (account.publicId == toID) {
+      throw ArgumentError("fromID and toID are the same");
+    }
+    fromIDName = account.name;
+  }
+
+  void setPairingMetadata(PairingMetadata pairingMetadata) {
+    this.pairingMetadata = pairingMetadata;
+  }
+
+  //Gets only the data stored here (in a dynamic format)
+  dynamic getData() {
+    return {fromID: fromID, toID: toID, amount: amount};
+  }
+
+  RequestSendQubicEvent({
+    required super.topic,
+    required super.requestId,
+    required this.fromID,
+    required this.toID,
+    required this.amount,
+  });
+
```

Comment:
many messages in this class are not localised

Reply:
The errors here are pushed to the WC client. Since this is a model I don't think we can have localization here. What do you suggest? @sallymoc